addons/knowledge/models/knowledge_article.py

```diff
@@ -202,7 +202,7 @@ def _compute_root_article_id(self):
                 parent = parent.parent_id
             articles.root_article_id = ancestors[-1:]
 
-    @api.depends('parent_id', 'internal_permission')
+    @api.depends('parent_id', 'parent_id.inherited_permission_parent_id', 'internal_permission')
```

Comment:
Indeed we do not read `inherited_permission_parent_id`, but is this an issue ?

I may be wrong, but the way I understand it is that adding `parent_id.inherited_permission_parent_id` in `@api.depends` will mark the fields computed using the method decorated as "needs to be recomputed if `inherited_permission_parent_id` of its `parent_id` changed".

Setting `parent.internal_permission` in the decorator does not work in our case (I tried) because of the recursion:

In that case, downgrading a permission on an article will force its child to recompute `inherited_permission_parent_id` and `inherited_permission`, but the `internal_permission` of the child won't change. Therefore the children of this child won't recompute `inherited_permission_parent_id`
and `inherited_permission`.

Setting `parent_id.inherited_permission_parent_id` instead makes sure that every child at any depth also recomputes their fields.

Reply:
And, understood concerning tests after a bugfix, I'll write some for the two fixes here and I will try to remember it :)

Reply:
Ok I think I understand, in that case it definitely deserves a comment:

```py
# "parent_id.inherited_permission_parent_id" is indirectly used, see method docstring for details
@api.depends('parent_id', 'parent_id.inherited_permission_parent_id', 'internal_permission')
def _compute_inherited_permission(self):
    """ Computed inherited internal permission. We go up ancestors until
    finding an article with an internal permission set, or a root article
    (without parent) or until finding a desynchronized article which
    serves as permission ancestor. Desynchronized articles break the
    permission tree finding.

    'parent_id.inherited_permission_parent_id' needs to be in the trigger
    as we will need to update this article's inherited permissions if our parent
    changes itself from which article it's inheriting.
    This allows cascading changes "downwards" when we modify the
    internal_permission of an article in the chain.

    It is however not directly used as we optimize the batching and group all
    articles by their parent_id. """
```

Feel free to modify / comment / trash if I misunderstood.

cc @dbeguin